service/history/execution/workflow.go

```diff
@@ -270,6 +281,19 @@ func (r *workflowImpl) terminateWorkflow(
 		WorkflowTerminationIdentity,
 	)
 
+	domainName := r.mutableState.GetDomainEntry().GetInfo().Name
+
+	logger.Info(
+		"Workflow execution terminated.",
+		tag.WorkflowDomainName(domainName),
+		tag.WorkflowID(r.mutableState.GetExecutionInfo().WorkflowID),
+		tag.WorkflowRunID(r.mutableState.GetExecutionInfo().RunID),
+		tag.WorkflowTerminationReason(WorkflowTerminationReason),
```

Comment:
is this code path only called for conflict based terminations?

Reply:
I recommend passing the reason as a parameter because this looks like a helper function that may get called from other places in the future. Having a reason param will force the future changes to pass the actual reason which is desired compared to logging the invalid reason for such cases.